### ISSUE TITLE
Thinking about the excursion set model definition, mean-free paths and recombination

### ISSUE THREAD

**Issue by daviesje** [21cmfast/21cmFAST#434]
The original excursion set algorithm defined the ionisation condition as a simple photon counting problem. If there are enough photons photons produced within a sphere to ionize the entire sphere, the entire sphere is ionised.

This interpretation gets somewhat muddied by the use of sharp-k filters and only flagging the central cell, which were found to both improve speed and accuracy compared to the statistics of radiative transfer simulation.

Davies et al 2022 introduced a filtering prescription which solidified the central cell algorithm as an alternate interpretation of the excursion set. That the filtered ionising emissivity fields represented the photons *reaching the central cell*, and the filter is used to directly account for attenuation of ionising photons on their way to the cell.

This interpretation has consequences for how we treat the ionisation barrier and recombinations, and some of these are naturally implemented in the new halo model, i.e the decoupling of the source and absorber fields. However, it also implies we should use the central cell's density and recombination rate to determine if a cell is ionised (in Davies+22 they still use the filtered delta and ignore recombinations).

The current implementation (when `flag_options.USE_EXP_FILTER==True and flag_options.CELL_RECOMB==True`) is as below, since it produces reasonable results:
- We still use the filtered density of the central cell in the ionisation condition
- However, recombinations are not filtered anymore, instead using the `dNrec` grid directly, this means the model uses a global MFP with the Davies+22 filter to account for attenuation outside the central cell, but recombinations internal to the cell are added to the ionisation barrier.

This implementation will require some discussion and testing. I have so far found that using the central cell density results in strange looking reionisation histories. We also need to make sure that our implementation doesn't double-count recombinations in the now separated attenuation and intra-cell absorption steps, as well as compare this approach to the regular `INHOMO_RECO` case.

In a nutshell, our existing `INHOMO_RECO` model accounts for the distribution of absorbers within a filtered region, but not the distribution of sources. The Davies+22 model accounts for the sources but not absorbers. I have attempted to put these two models together so we take into account both, however we should make sure that what we are doing is reasonable.
